sway/input/cursor.c

```diff
@@ -162,7 +162,29 @@ void cursor_send_pointer_motion(struct sway_cursor *cursor, uint32_t time_msec,
 				seat_set_focus_warp(cursor->seat, c, false);
 			}
 		} else {
-			seat_set_focus_warp(cursor->seat, c, false);
+			// Don't switch focus on title mouseover for
+			// stacked and tabbed layouts
+			// If pointed container is in nested containers which are
+			// inside tabbed/stacked layout we should skip them
+			bool do_mouse_focus = true;
+			struct sway_container *p = c->parent;
+			while (p) {
+				if ((p->layout == L_TABBED || p->layout == L_STACKED)
+					&& !view_is_visible(c->sway_view)) {
```

Comment:
Move the call to `view_is_visible` outside of the while loop and store it in a variable rather than recomputing it on each iteration of the loop.

Also, assuming that `c` is a view makes me a little nervous. Please add an assertion for this.